autogpt_platform/backend/backend/data/rabbitmq.py

```diff
@@ -0,0 +1,232 @@
+import logging
+import os
+from typing import Optional, Callable, Any
+
+from dotenv import load_dotenv
+import pika
+from pika.adapters.blocking_connection import BlockingChannel
+from pika.spec import Basic, BasicProperties
+import aio_pika
+from aio_pika.abc import (
+    AbstractChannel,
+    AbstractConnection,
+    AbstractQueue,
+    AbstractIncomingMessage,
+)
+
+from backend.util.retry import conn_retry
+
+load_dotenv()
+
+HOST = os.getenv("RABBITMQ_HOST", "localhost")
+PORT = int(os.getenv("RABBITMQ_PORT", "5672"))
+USERNAME = os.getenv("RABBITMQ_DEFAULT_USER", "guest")
+PASSWORD = os.getenv("RABBITMQ_DEFAULT_PASS", "guest")
+VIRTUAL_HOST = os.getenv("RABBITMQ_VHOST", "/")
+
+logger = logging.getLogger(__name__)
+
+# Synchronous connections
+connection: Optional[pika.BlockingConnection] = None
+channel: Optional[BlockingChannel] = None
+
+# Async connections
+connection_async: Optional[AbstractConnection] = None
+channel_async: Optional[AbstractChannel] = None
+
+
+@conn_retry("RabbitMQ", "Acquiring connection")
+def connect() -> pika.BlockingConnection:
+    global connection, channel
```

Comment:
Can we avoid using globals please. It causes a lot of issues with state management and debugging.

It took ages to successful extract globals from autogpt classic, if you recall